scripts/src/main/scala/com/gu/mediaservice/scripts/EsScript.scala

```diff
@@ -99,13 +104,22 @@ object Reindex extends EsScript {
 
         Thread.sleep(1000)
 
-        reindexScroll(query.execute.actionGet)
+        if(fromTime.isEmpty){
+          reindexScroll(query.setQuery(matchAllQuery).execute.actionGet)
+        } else {
+          val recentChangesQuery = rangeQuery("lastModified").from(fromTime.get).to(DateTime.now)
+          reindexScroll(query.setQuery(recentChangesQuery).execute.actionGet)
+        }
 
         // TODO: deleteIndex when we are confident
         client.close()
       }
     }
-    EsClient.reindex
+    if(extraArgs.isEmpty) {
```

Comment:
I'd avoid `if` statements like this wherever possible and either `foldLeft` or return a value:

```scala
val date = if(extraArgs.isEmpty) None else Some((DateTime.parse(extraArgs(0))))
EsClient.reindex(date)
```

or

```scala
EsClient.reindex(
  extraArgs.foldLeft(None)(Some(DateTime.parse(extraArgs(0))))
)
```